src/olympia/addons/tests/test_admin.py

```diff
@@ -53,35 +50,114 @@ def test_invalid_urls(self):
+        assert ReplacementAddon.objects.filter(pk=replacement.pk).exists()
+
+    def test_can_edit_with_content_curate_permission(self):
+        replacement = ReplacementAddon.objects.create(
+            guid='@foo', path='/addon/foo-replacement/')
+        self.detail_url = reverse(
+            'admin:addons_replacementaddon_change', args=(replacement.pk,)
+        )
+        user = user_factory()
+        self.grant_permission(user, 'Content:Curate')
+        self.client.login(email=user.email)
+        response = self.client.get(self.detail_url, follow=True)
+        assert response.status_code == 200
+        assert '/addon/foo-replacement/' in response.content
+
+        response = self.client.post(
+            self.detail_url, {'guid': '@bar', 'path': replacement.path},
+            follow=True)
+        assert response.status_code == 200
+        replacement.reload()
+        assert replacement.guid == '@bar'
+
+    def test_can_delete_with_content_curate_permission(self):
+        replacement = ReplacementAddon.objects.create(
+            guid='@foo', path='/addon/foo-replacement/')
+        self.delete_url = reverse(
+            'admin:addons_replacementaddon_delete', args=(replacement.pk,)
+        )
+        user = user_factory()
+        self.grant_permission(user, 'Content:Curate')
+        self.client.login(email=user.email)
+        response = self.client.get(self.delete_url, follow=True)
+        assert response.status_code == 200
+        response = self.client.post(
+            self.delete_url, data={'post': 'yes'}, follow=True)
+        assert response.status_code == 200
+        assert not ReplacementAddon.objects.filter(pk=replacement.pk).exists()
+
+    def test_can_list_with_content_curate_permission(self):
+        user = user_factory()
+        self.grant_permission(user, 'Content:Curate')
+        self.client.login(email=user.email)
         # '@foofoo&foo' isn't a valid guid, because &, but testing urlencoding.
         ReplacementAddon.objects.create(guid='@foofoo&foo', path='/addon/bar/')
-        request = RequestFactory().get(
-            '/en-US/admin/models/addons/replacementaddon/')
-        request.user = UserProfile.objects.get(email='admin@mozilla.com')
-        adminview = ReplacementAddonAdmin(
-            ReplacementAddon, StaffAdminSite(name='staffadmin'))
-        view = adminview.changelist_view(request)
-        assert '@foofoo&amp;foo' in view.rendered_content
-        assert '/addon/bar/' in view.rendered_content
-        test_url = '<a href="%s">Test</a>' % (
-            reverse('addons.find_replacement') + '?guid=%40foofoo%26foo')
-        assert test_url in view.rendered_content, view.rendered_content
+        response = self.client.get(self.list_url, follow=True)
+        assert response.status_code == 200
+        assert '@foofoo&amp;foo' in response.content
+        assert '/addon/bar/' in response.content
+        test_url = str('<a href="%s">Test</a>' % (
```

Comment:
`str` instead of `unicode`?

Reply:
`response.content` is a `str`, not `unicode`. The alternative is leaving `test_url` alone here and writing `test_url in response.content.decode('utf-8')` below which may be clearer?

Reply:
I don't mind, I just thought it unusual.